### PR TITLE
fix: targeting ui dir structure

### PR DESCRIPTION
**Pull request by pandeymangg**
## What does this PR do?

fixes [formbricks/internal#215](https://github.com/formbricks/internal/issues/215), moves the UI components from the `ui/Targeting` directory to their individual directories.

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/ui/BasicSegmentFilter/index.tsx
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/index.tsx:1]: `BasicSegmentFilter` is used only in the UI package itself and no where else`ui/BasicSegmentEditor`, so can we move it and its corresponding files and folders into `ui/BasicSegmentEditor`, so it looks like this

Screenshot 2024-06-03 at 4 28 12 PM

File: packages/ui/BasicSegmentFilter/components/AttributeSegmentFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  TSegmentConnector,
+  TSegmentFilterValue,
+} from "@formbricks/types/segment";
+
+import { Input } from "../../Input";
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "../../Select";
+import { SegmentFilterItemConnector } from "./SegmentFilterItemConnector";
+import { SegmentFilterItemContextMenu } from "./SegmentFilterItemContextMenu";
+
+type TAttributeSegmentFilterProps = {
```
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/components/AttributeSegmentFilter.tsx:30]: type -> interface

File: packages/ui/BasicSegmentFilter/components/PersonSegmentFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  TSegmentFilterValue,
+  TSegmentPersonFilter,
+} from "@formbricks/types/segment";
+
+import { Input } from "../../Input";
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "../../Select";
+import { SegmentFilterItemConnector } from "./SegmentFilterItemConnector";
+import { SegmentFilterItemContextMenu } from "./SegmentFilterItemContextMenu";
+
+type TPersonSegmentFilterProps = {
```
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/components/PersonSegmentFilter.tsx:28]: type -> interface

File: packages/ui/BasicSegmentFilter/components/SegmentFilterItemConnector.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { TSegment, TSegmentConnector } from "@formbricks/types/segment";
+
+export const SegmentFilterItemConnector = ({
+  connector,
+  segment,
+  setSegment,
+  filterId,
+  viewOnly,
+}: {
+  connector: TSegmentConnector;
```
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/components/SegmentFilterItemConnector.tsx:12]: Can we add interface `SegmentFilterItemConnectorProps`

File: packages/ui/BasicSegmentFilter/components/SegmentFilterItemContextMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import { Button } from "../../Button";
+import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from "../../DropdownMenu";
+
+export const SegmentFilterItemContextMenu = ({
+  filterId,
+  onDeleteFilter,
+  onMoveFilter,
+  viewOnly,
+}: {
+  filterId: string;
```
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/components/SegmentFilterItemContextMenu.tsx:14]: can we add interface `SegmentFilterItemContextMenuProps` 😊

File: packages/ui/BasicSegmentFilter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  TSegmentAttributeFilter,
+  TSegmentConnector,
+  TSegmentFilter,
+  TSegmentPersonFilter,
+} from "@formbricks/types/segment";
+
+import { AttributeSegmentFilter } from "./components/AttributeSegmentFilter";
+import { PersonSegmentFilter } from "./components/PersonSegmentFilter";
+
+type TBasicSegmentFilterProps = {
```
Comment (Dhruwang) [inline packages/ui/BasicSegmentFilter/index.tsx:15]: type -> interface

**Review by Dhruwang — CHANGES_REQUESTED** [review]
Hey @pandeymangg , looks pretty solid 🚀🙌🏻💪.
Just some small comments regarding relocation. Also can you check and replace `type` with `interface` in relocated components 😊

File: packages/lib/utils.ts [hunk 6]
```diff
@@ -83,3 +83,5 @@ export const isLight = (color: string) => {
   }
   return r * 0.299 + g * 0.587 + b * 0.114 > 128;
 };
+
+export const isCapitalized = (str: string) => str.charAt(0) === str.charAt(0).toUpperCase();
```
Comment (Dhruwang) [inline packages/lib/utils.ts:87]: Just a suggestion, We have some util functions related to string in `lib/strings.ts`, but that too feels a bit odd. So can we move `strings.ts` to` lib/utils` folder and also include this function there. What do you think 🤔

Comment (pandeymangg) [reply packages/lib/utils.ts:87]: sounds good, this file also has some functions for colors, so maybe we can also have a `colors.ts` inside the utils dir with those functions

**Review by Dhruwang — APPROVED** [review]
Good to go ✅🚀

**Pull request merged**